tools/testsys/src/run.rs

```diff
@@ -0,0 +1,230 @@
+use crate::aws_resources::AwsK8s;
+use anyhow::{anyhow, ensure, Context, Result};
+use bottlerocket_types::agent_config::TufRepoConfig;
+use buildsys::Variant;
+use clap::Parser;
+use model::test_manager::TestManager;
+use model::SecretName;
+use pubsys_config::InfraConfig;
+use serde::Deserialize;
+use serde_plain::derive_fromstr_from_deserialize;
+use std::collections::HashMap;
+use std::env;
+use std::fs::File;
+use std::path::PathBuf;
+
+/// Run a set of tests on a given arch and variant
+#[derive(Debug, Parser)]
+pub(crate) struct Run {
+    /// The type of test to run
+    test_flavor: TestType,
+
+    /// The arch we to test
+    #[clap(long, env = "BUILDSYS_ARCH")]
+    arch: String,
+
+    /// The variant we to test
+    #[clap(long, env = "BUILDSYS_VARIANT")]
+    variant: String,
+
+    /// The path to `infra.toml`
+    #[clap(long, env = "PUBLISH_INFRA_CONFIG_PATH", parse(from_os_str))]
+    infra_config_path: PathBuf,
+
+    /// The name of the repo we want for upgrade/downgrade testing.
+    /// The variant we to test
+    #[clap(long, env = "PUBLISH_REPO")]
+    repo: String,
+
+    /// The path to `amis.json`
+    #[clap(long)]
+    ami_input: String,
+
+    /// Override for the region the tests should be run in. If none are provided the first region in
+    /// infra will be used.
+    #[clap(long, env = "TESTSYS_REGION")]
+    region: Option<String>,
+
+    /// The custom kube conformance image that should be used by sonobuoy.
+    #[clap(long)]
+    kube_conformance_image: Option<String>,
```

Comment:
Hmm this is challenging. We're taking an argument here that is specific to a certain variant and test mode.